one/src/migrations.rs

```diff
@@ -150,6 +158,85 @@ impl BlockStore for FSBlockStore {
     }
 }
 
+impl FSBlockStore {
+    fn blocks_from_dir(&self) -> BoxStream<'static, anyhow::Result<(Cid, Vec<u8>)>> {
+        // the block store is split in to 1024 directories and then the blocks stored as files.
+        // the dir structure is the penultimate two characters as dir then the b32 sha256 multihash of the block
+        // The leading "B" for the b32 sha256 multihash is left off
+        // ~/.ipfs/blocks/QV/CIQOHMGEIKMPYHAUTL57JSEZN64SIJ5OIHSGJG4TJSSJLGI3PBJLQVI.data // cspell:disable-line
+        info!(path = %self.input_ipfs_path.display(), "opening IPFS repo");
+
+        let mut dirs = Vec::new();
+        dirs.push(self.input_ipfs_path.clone());
+
+        let offset = self.file_offset;
```

Comment:
I don't this we should support limit/offset for the read dir case. From the docs:
 
> The order in which this iterator returns entries is platform and filesystem dependent.

We cannot guarantee that limit/offset will be meaningful. And in practice the order in which blocks where traversed appeared to be random. Meaning it was not in lexicographical order on my local filesystem nor in various production filesystems we used.

I propose if limit/offset are not None and there is no input file list then we error.

Reply:
https://doc.rust-lang.org/nightly/std/fs/fn.read_dir.html

Reply:
I will likely be nearly stable on any one disk as the tree in the filesystem try's to not change much. If the user wants a guaranty that they cover the entire ipfs repo then they should use make a file list file.
I wander if we should make a CLI argument for making a sorted file list file and a CLI for subtracting one from another

or maybe in the make file
```
# List files in directory at migration_1 and sort them alphabetically
ls -l ipfs/blocks | sort > migration_1.txt
 
CERAMIC_ONE_INPUT_FILE_LIST_PATH=migration_1.txt \
ceramic-one migrations from-ipfs 

# List files in directory at migration_2 and sort them alphabetically
ls -l ipfs/blocks | sort > migration_2.txt

# Find files in migration_2 that are not in migration_1
comm -23  migration_2.txt migration_1.txt > delta

CERAMIC_ONE_INPUT_FILE_LIST_PATH=delta.txt \
ceramic-one migrations from-ipfs 
```

Reply:
Do we need the feature? I'd rather not add an unused feature that has a potential to break in mysterious ways.

Reply:
I took it out for now and added the check.